### PR TITLE
Handle array of paths as route pattern

### LINKED ISSUE #2892: [feature]: Path array for venia-ui routes pattern
**Is your feature request related to a problem? Please describe.**
Passing an array of paths to the `pattern` field of venia-ui route definition does not work. This value is passed as a string value in BabelRouteInjectionPlugin.js (86:76). It should also accept an array, as react-router path does (see [docs](https://reactrouter.com/web/api/Route/path-string-string)).

**Describe the solution you'd like**

```
veniaTargets.routes.tap(routes => [
        {
            name: 'Home Page',
            pattern: ['/home', '/'],
            exact: true,
            path: '@components/HomePage'
        },
    ]);
```

**Please let us know what packages this feature is in regards to:**
- [ ] `venia-concept`
- [x] `venia-ui`
- [ ] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by papiliond**
## Description

Extend the functionality of the Routes target of venia-ui. Handle both `string` and `string[]` type of the react-router path prop (see [docs](https://reactrouter.com/web/api/Route/path-string-string)).

## Related Issue
Closes #2892

### Verification Steps

1. Add/modify an entry in "packages/venia-ui/lib/defaultRoutes.json" and define the `pattern` with an array value. For example:
```
    {
        "name": "AccountInformationPage",
        "pattern": ["/account-information", "/acc-info"],
        "exact": true,
        "path": "../AccountInformationPage"
    }
```
2. Run the project with `yarn watch`
3. Navigating to "/account-information" and "/acc-info" gives you the same page

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/targets/makeRoutesTarget.js [hunk 1]
```diff
@@ -11,10 +11,14 @@ function makeRoutesTarget(venia) {
...
+
+        const routePath = Array.isArray(route.pattern) ? 
+        `{[${route.pattern.map(path => `"${path}"`).join(", ")}]}` : `"${route.pattern}"`;
+
         routeList.prependJSX(
             'Switch',
             `<Route ${route.exact ? 'exact ' : ''}path="${
-                route.pattern
+                routePath
             }"><${AddedRoute}/></Route>`
```
Comment (sirugh) [inline packages/venia-ui/lib/targets/makeRoutesTarget.js:22]: The `routePath` creation now adds the double-quotes or brackets when necessary so you should remove them from line 20/22.

File: packages/venia-ui/lib/targets/makeRoutesTarget.js [hunk 2]
```diff
@@ -11,10 +11,14 @@ function makeRoutesTarget(venia) {
 function addRoutes(routeList, routes) {
     for (const route of routes) {
         const AddedRoute = routeList.addReactLazyImport(route.path, route.name);
+
+        const routePath = Array.isArray(route.pattern) ? 
+        `{[${route.pattern.map(path => `"${path}"`).join(", ")}]}` : `"${route.pattern}"`;
```
Comment (sirugh) [inline packages/venia-ui/lib/targets/makeRoutesTarget.js:16]: `JSON.stringify` works as well and is a bit easier to read.

```suggestion
        `{[${JSON.stringify(route.pattern}]}` : `"${route.pattern}"`;
```

**Review by sirugh — CHANGES_REQUESTED** [review]
Would you mind adding tests for the cases we are supporting here? Two simple assertions for the expected string in both cases would suffice. You'll also want to make sure to check on the failing test.

Also please make sure to run `yarn prettier` and potentially `yarn lint` to fix any lint errors.

**Comment by papiliond** [comment]
@sirugh Thanks for answering. Will create tests and review your suggestions this weekend.

**Comment by papiliond** [comment]
@sirugh Made some refactors and added tests for the two cases.

**Review by sirugh — APPROVED** [review]

**Comment by dpatil-magento** [comment]
QA Approved.

**Pull request merged**